### DISCUSSION TITLE
`@/lib/utils` is hijacked?

### DISCUSSION THREAD

**Discussion by gera2ld** [unovue/shadcn-vue#902] [Q&A]
It seems that `@/lib/utils` is hijacked and hard-coded as an alias to `.nuxt/shadcn-next/utils.ts`, despite that a `lib/utils.ts` is created on project initialization. As a result, any changes to `lib/util.ts` are ignored.

This is so counter-intuitive! Am I not allowed to use `@/lib/utils`?

https://github.com/unovue/shadcn-vue/blob/b7ef4653f75ea30b28ec2ff635cda4ab71dc2984/packages/module/src/module.ts#L38-L43

**Comment by metinkahraman-github** [comment]
I have the same problem.

**_500  The requested module '/_nuxt/@fs/C:/Projects/dev/newappv2/.nuxt/shadcn-nuxt/utils.ts' does not provide an export named 'valueUpdater'_**

**Original @/lib/utils.ts**
```
import { type ClassValue, clsx } from 'clsx'
import { twMerge } from 'tailwind-merge'

import type { Updater } from '@tanstack/vue-table'
import type { Ref } from 'vue'

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs))
}

export function valueUpdater<T extends Updater<unknown>>(updaterOrValue: T, ref: Ref) {
  ref.value = typeof updaterOrValue === 'function'
    ? updaterOrValue(ref.value)
    : updaterOrValue
}
```

**.nuxt/shadcn-nuxt/utils.ts**
```
import { type ClassValue, clsx } from 'clsx'
import { twMerge } from 'tailwind-merge'

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs))
}
```

**Comment by metinkahraman-github** [comment]
@gera2ld 
When I used '~' instead of '@' when importing, the error disappeared. I will use it like this until it is fixed.

`import { valueUpdater } from '~/lib/utils'`

**Reply by gera2ld** [reply]
Thanks for sharing. This is also my workaround. Luckily they only hard-coded `@/lib/utils` in their code.
But I'm still surprised to see a normal file being hijacked like this.

**Reply by Pierre-Assemat** [reply]
Any updates on that ?

**Comment by szulcus** [comment]
I have the same problem

**Comment by sadeghbarati** [comment]
I will remove the utils content from the module package so that the package has a single purpose: to add auto-imported components that Nuxt cannot scan

**Reply by hoanghjx** [reply]
Is there an update?

**Reply by sadeghbarati** [reply]
Hi @hoanghjx module is ready, it only need be published to the npm
